crates/net/downloaders/src/headers/linear.rs

```diff
@@ -548,6 +575,18 @@ where
     fn poll_next(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
         let this = self.get_mut();
 
+        // The downloader boundaries (local head and sync target) have to be set in order
+        // to start downloading data.
+        if this.local_head.is_none() || this.sync_target.is_none() {
+            tracing::warn!(
```

Comment:
this can be trace!

we'll definitely hit that since downloader is spawned, it will immediately be pending and the first `update_sync_target` message will awake it

Reply:
agreed